### DISCUSSION TITLE
Need help with patching Comic Code - Incorrent glyph rendering.

### DISCUSSION THREAD

**Discussion by EmreDogann** [ryanoasis/nerd-fonts#1298] [General]
Hey there,

I use [Comic Code](https://tosche.net/fonts/comic-code) which is a paid font that I patched.

Using the fontpatcher, I call this command to patch the fonts:
`fontforge -script ~/FontPatcher/font-patcher "/mnt/c/Fonts/Comic Code/Nerd Font Patched/ComicCode-Regular.otf" --out "/mnt/c/Fonts/Comic Code/Nerd Font Patched" --makegroup -1 --complete`

This seems to work mostly fine, doesn't have any issues with missing glyphs, however the problems are with how the glyphs are rendered:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/a6c580bd-728b-4836-a08b-6e3b794f13ec)

You can see that glyph U+E0B6 (1) renders fine, but glyph U+EB04 (2) is cutoff slightly on its right edge.

But my bigger problem is that glyphs don't seem to be taking up the full height of the character box:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/69196de6-a892-420b-87a9-26cf4817e4dc)

This is in Neovim, the issue is that the vertical bars being rendered (U+2502) are supposed to be full height, but instead the top of the glyph is cutoff, leading to that dashed line. Using the pre-patched CaskaydiaCove Nerd Font, you can see what the glyph is supposed to look like:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/68248fa8-5b71-4ffb-ae32-7380b895dffe)

If I use Windows Terminal Preview version 1.18.1462.0, I can change the line height to 1.2 which fixes the vertical lines, however now the text is too close together, making it difficult to read for me:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/de900b35-8bfc-4ca5-ac33-e46a352359c3)

Is this a problem with the way I'm patching the font, or is there just nothing I can do in this scenario?

### Environment:
Windows 11, WSL 2, Ubuntu
Windows Terminal 1.17.11461.0
NerdFont Patcher 3.0.2 (4.4.0)

**Comment by Finii** [comment]
The way you patch is fine.
The effect you see with [2] should not be there.

I think you do not want `--mono` but maybe you can try if the problem is still there if you use the `Nerd Font Mono` variant.
But than, it already looks like a NFM font, the Lua icon is so small?
<img width="261" alt="Screenshot 2023-06-14 at 06 57 22" src="https://github.com/ryanoasis/nerd-fonts/assets/16012374/69c17b4b-7fdd-4031-b0b3-bff7abfa954e">


And then, this is WSL, which has its own problems. Maybe just use plain Windows, i.e. install the font on Windows and use the Windows Terminal for Powershell or CMD or whatever, on the Windows machine.

If I remember correctly I did patch that font once and there was no problem.
Maybe you can share the output of a patch run with added `--debug 2`?

> I can change the line height to 1.2 which fixes the vertical lines, however now the text is too close together

Hmm? That means before you had a even bigger line spacing like 1.3 or something?

**Reply by EmreDogann** [reply]
> But than, it already looks like a NFM font, the Lua icon is so small?

Yeah I've tried it will have variants of the patcher argument flags but the Icons always seems a little too small (and sometimes not centered correctly).

> Maybe you can share the output of a patch run with added --debug 2?

Below is the output using the same command from the original post but with `--debug 2`:

```shell
Copyright (c) 2000-2022. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20201107
 Based on sources from 2022-01-13 19:13 UTC-ML-D-GDK3.
PythonUI_Init()
copyUIMethodsToBaseTable()
Nerd Fonts Patcher v3.0.2 (4.4.0) (ff 20201107)
DEBUG: Naming mode -1
DEBUG: Monospace check: Panose is invalid ([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]); glyph-width-mono True
INFO: Setting Panose 'Family Kind' to 'Latin Text and Display' (was 'Any')
INFO: Setting Panose 'Proportion' to 'Monospaced' (was 'Any')
WARNING: Font vertical metrics probably wrong USE TYPO METRICS, assume opposite (i.e. False)
DEBUG: Extended glyphs wider bounding box than basic glyphs
DEBUG: Font has negative right side bearing in extended glyphs
DEBUG: Final font cell dimensions 620 w x 1467 h
Adding 180 Glyphs from Seti-UI + Custom Set
╢████████████████████████████████████████╟ 100%
Adding 6 Glyphs from Heavy Angle Brackets Set
╢████████████████████████████████████████╟ 100%
Adding 198 Glyphs from Devicons Set
╢████████████████████████████████████████╟ 100%
Adding 3 Glyphs from Powerline Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 4 Glyphs from Powerline Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 1 Glyphs from Powerline Extra Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 21 Glyphs from Powerline Extra Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 1 Glyphs from Powerline Extra Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 8 Glyphs from Powerline Extra Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 1 Glyphs from Powerline Extra Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 11 Glyphs from Pomicons Set
╢████████████████████████████████████████╟ 100%
The following table(s) in the font have been ignored by FontForge
  Ignoring 'EPAR'
The glyph named asterisk is mapped to U+F069.
  But its name indicates it should be mapped to U+002A.
The glyph named plus is mapped to U+F067.
  But its name indicates it should be mapped to U+002B.
The glyph named question is mapped to U+F128.
  But its name indicates it should be mapped to U+003F.
The glyph named minus is mapped to U+F068.
  But its name indicates it should be mapped to U+2212.
The glyph named heart is mapped to U+F004.
  But its name indicates it should be mapped to U+2665.
The glyph named home is mapped to U+F015.
  But its name indicates it should be mapped to U+21B8.
The glyph named check is mapped to U+F046.
  But its name indicates it should be mapped to U+2713.
The glyph named bell is mapped to U+F0A2.
  But its name indicates it should be mapped to U+2407.
The glyph named lessequal is mapped to U+F500.
  But its name indicates it should be mapped to U+2264.
The glyph named circle is mapped to U+F111.
  But its name indicates it should be mapped to U+25CB.
The glyph named smile is mapped to U+F118.
  But its name indicates it should be mapped to U+263A.
The glyph named frown is mapped to U+F119.
  But its name indicates it should be mapped to U+2322.
The glyph named bullseye is mapped to U+F140.
  But its name indicates it should be mapped to U+25CE.
The glyph named compass is mapped to U+F14E.
  But its name indicates it should be mapped to U+263C.
The glyph named female is mapped to U+F182.
  But its name indicates it should be mapped to U+2640.
The glyph named male is mapped to U+F183.
  But its name indicates it should be mapped to U+2642.
The glyph named sun is mapped to U+F185.
  But its name indicates it should be mapped to U+263C.
The glyph named venus is mapped to U+F221.
  But its name indicates it should be mapped to U+2640.
Adding 679 Glyphs from Font Awesome Set
╢████████████████████████████████████████╟ 100%
The glyph named slash is mapped to U+E016.
  But its name indicates it should be mapped to U+002F.
The glyph named pi is mapped to U+E02C.
  But its name indicates it should be mapped to U+03C0.
The glyph named ring is mapped to U+E03D.
  But its name indicates it should be mapped to U+02DA.
The glyph named infinity is mapped to U+E055.
  But its name indicates it should be mapped to U+221E.
The glyph named equal is mapped to U+E079.
  But its name indicates it should be mapped to U+003D.
Adding 170 Glyphs from Font Awesome Extension Set
╢████████████████████████████████████████╟ 100%
Adding 4 Glyphs from Power Symbols Set
╢████████████████████████████████████████╟ 100%
Adding 1 Glyphs from Power Symbols Set
╢████████████████████████████████████████╟ 100%
The glyph named alpha is mapped to U+F002B.
  But its name indicates it should be mapped to U+03B1.
The glyph named at is mapped to U+F0065.
  But its name indicates it should be mapped to U+0040.
The glyph named backspace is mapped to U+F006E.
  But its name indicates it should be mapped to U+2408.
The glyph named bell is mapped to U+F009A.
  But its name indicates it should be mapped to U+2407.
The glyph named beta is mapped to U+F00A1.
  But its name indicates it should be mapped to U+03B2.
The glyph named box is mapped to U+F00D4.
  But its name indicates it should be mapped to U+25A1.
The glyph named check is mapped to U+F012C.
  But its name indicates it should be mapped to U+2713.
The glyph named compass is mapped to U+F018B.
  But its name indicates it should be mapped to U+263C.
The glyph named delete is mapped to U+F01B4.
  But its name indicates it should be mapped to U+2421.
The glyph named delta is mapped to U+F01C2.
  But its name indicates it should be mapped to U+03B4.
The glyph named earth is mapped to U+F01E7.
  But its name indicates it should be mapped to U+2641.
The glyph named equal is mapped to U+F01FC.
  But its name indicates it should be mapped to U+003D.
The glyph named heart is mapped to U+F08D0.
  But its name indicates it should be mapped to U+2665.
The glyph named home is mapped to U+F02DC.
  But its name indicates it should be mapped to U+21B8.
The glyph named image is mapped to U+F02E9.
  But its name indicates it should be mapped to U+22B7.
The glyph named minus is mapped to U+F0374.
  But its name indicates it should be mapped to U+2212.
The glyph named omega is mapped to U+F03C9.
  But its name indicates it should be mapped to U+03C9.
The glyph named percent is mapped to U+F03F0.
  But its name indicates it should be mapped to U+0025.
The glyph named pi is mapped to U+F03FF.
  But its name indicates it should be mapped to U+03C0.
The glyph named plus is mapped to U+F0415.
  But its name indicates it should be mapped to U+002B.
The glyph named sigma is mapped to U+F04A0.
  But its name indicates it should be mapped to U+03C3.
The glyph named tie is mapped to U+F0519.
  But its name indicates it should be mapped to U+2040.
The glyph named triangle is mapped to U+F0536.
  But its name indicates it should be mapped to U+25B3.
The glyph named bullseye is mapped to U+F05DD.
  But its name indicates it should be mapped to U+25CE.
The glyph named copyright is mapped to U+F05E6.
  But its name indicates it should be mapped to U+00A9.
The glyph named lambda is mapped to U+F0627.
  But its name indicates it should be mapped to U+03BB.
The glyph named asterisk is mapped to U+F06C4.
  But its name indicates it should be mapped to U+002A.
The glyph named hook is mapped to U+F06E2.
  But its name indicates it should be mapped to U+2440.
The glyph named infinity is mapped to U+F06E4.
  But its name indicates it should be mapped to U+221E.
The glyph named prescription is mapped to U+F0706.
  But its name indicates it should be mapped to U+211E.
The glyph named tilde is mapped to U+F0725.
  But its name indicates it should be mapped to U+02DC.
The glyph named cancel is mapped to U+F073A.
  But its name indicates it should be mapped to U+2418.
The glyph named square is mapped to U+F0764.
  But its name indicates it should be mapped to U+25A1.
The glyph named null is mapped to U+F07E2.
  But its name indicates it should be mapped to U+2400.
The glyph named pipe is mapped to U+F07E5.
  But its name indicates it should be mapped to U+01C0.
The glyph named ring is mapped to U+F07EB.
  But its name indicates it should be mapped to U+02DA.
The glyph named cross is mapped to U+F0953.
  But its name indicates it should be mapped to U+2629.
The glyph named trademark is mapped to U+F0A78.
  But its name indicates it should be mapped to U+2122.
The glyph named ampersand is mapped to U+F0A8D.
  But its name indicates it should be mapped to U+0026.
The glyph named diamond is mapped to U+F0B8A.
  But its name indicates it should be mapped to U+2666.
The glyph named bullet is mapped to U+F0CF3.
  But its name indicates it should be mapped to U+2022.
The glyph named comma is mapped to U+F0E23.
  But its name indicates it should be mapped to U+002C.
The glyph named rectangle is mapped to U+F0E5E.
  But its name indicates it should be mapped to U+25AD.
The glyph named spade is mapped to U+F0E65.
  But its name indicates it should be mapped to U+2660.
The glyph named epsilon is mapped to U+F10E0.
  But its name indicates it should be mapped to U+03B5.
The glyph named gamma is mapped to U+F10EE.
  But its name indicates it should be mapped to U+03B3.
Adding 6896 Glyphs from Material Set
╢████████████████████████████████████████╟ 100%
The following table(s) in the font have been ignored by FontForge
  Ignoring 'webf'
Adding 228 Glyphs from Weather Icons Set
╢████████████████████████████████████████╟ 100%
Adding 48 Glyphs from Font Logos Set
╢████████████████████████████████████████╟ 100%
The glyph named question is mapped to U+F02C.
  But its name indicates it should be mapped to U+003F.
The glyph named check is mapped to U+F03A.
  But its name indicates it should be mapped to U+2713.
The glyph named plus is mapped to U+F05D.
  But its name indicates it should be mapped to U+002B.
The glyph named x is mapped to U+F081.
  But its name indicates it should be mapped to U+0078.
The glyph named home is mapped to U+F08D.
  But its name indicates it should be mapped to U+21B8.
The glyph named ellipsis is mapped to U+F09A.
  But its name indicates it should be mapped to U+2026.
The glyph named bell is mapped to U+F0DE.
  But its name indicates it should be mapped to U+2407.
The glyph named circle is mapped to U+F27D.
  But its name indicates it should be mapped to U+25CB.
The glyph named diamond is mapped to U+F292.
  But its name indicates it should be mapped to U+2666.
The glyph named image is mapped to U+F2B8.
  But its name indicates it should be mapped to U+22B7.
The glyph named infinity is mapped to U+F2B9.
  But its name indicates it should be mapped to U+221E.
The glyph named square is mapped to U+F2F0.
  But its name indicates it should be mapped to U+25A1.
The glyph named sun is mapped to U+F2F5.
  But its name indicates it should be mapped to U+263C.
Adding 169 Glyphs from Octicons Set
╢████████████████████████████████████████╟ 100%
Adding 1 Glyphs from Octicons Set
╢████████████████████████████████████████╟ 100%
Adding 1 Glyphs from Octicons Set
╢████████████████████████████████████████╟ 100%
Adding 139 Glyphs from Octicons Set
╢████████████████████████████████████████╟ 100%
The glyph named bell is mapped to U+EAA2.
  But its name indicates it should be mapped to U+2407.
The glyph named check is mapped to U+EAB2.
  But its name indicates it should be mapped to U+2713.
The glyph named compass is mapped to U+EBD5.
  But its name indicates it should be mapped to U+263C.
The glyph named ellipsis is mapped to U+EA7C.
  But its name indicates it should be mapped to U+2026.
The glyph named heart is mapped to U+EB05.
  But its name indicates it should be mapped to U+2665.
The glyph named home is mapped to U+EB06.
  But its name indicates it should be mapped to U+21B8.
The glyph named newline is mapped to U+EBEA.
  But its name indicates it should be mapped to U+2424.
The glyph named question is mapped to U+EB32.
  But its name indicates it should be mapped to U+003F.
Adding 388 Glyphs from Codicons Set
╢████████████████████████████████████████╟ 100%
Done with Patch Sets, generating font...
WARNING: Renaming disabled! Make sure to comply with font license, esp RFN clause!
DEBUG: Tweaking 1/1
   Comic Code Regular
   \===> '/mnt/c/Fonts/Comic Code/Nerd Font Patched/ComicCode-Regular.otf'
``` 

> And then, this is WSL, which has its own problems. Maybe just use plain Windows, i.e. install the font on Windows and use the Windows Terminal for Powershell or CMD or whatever, on the Windows machine.

Hmm I'm not sure I understand? I have the font installed on Windows and then set the font in the Windows Terminal settings. I haven't touched any font configuration on the WSL side.

> Hmm? That means before you had a even bigger line spacing like 1.3 or something?

In the release version of Windows Terminal they do no expose the line height so I unfortunately don't know what value they set it to. In the preview releases they expose it next to the font selection.

**Comment by Finii** [comment]
The first two images, which terminal is that?
If you say neovim, what do you mean by that. I also use neovim (https://neovim.io), but that is a command line application.

If you try to increase the line hight (line to line distance), there are some terminals out there that stretch glyphs that are "a full cell height", but of course do not touch glyphs that are not touching top and bottom of the cell. Depending on which codepoint you use for the vertical bar, maybe that glyph is not reaching the top and bottom. If that is not patched in by the patcher we have no influence on that. Furthermore usually only the Powerline glyphs and the block building glyphs are scaled for full height by the patcher.
But that is all about the vertical bar. The cut-off Powerline symbol is clearly a bug of sorts.

**Reply by EmreDogann** [reply]
> The first two images, which terminal is that?
If you say neovim, what do you mean by that. I also use neovim ([neovim.io](https://neovim.io/)), but that is a command line application.

All the images are from Windows Terminal, running a WSL 2 Ubuntu instance. In that, I then have a neovim installation where I do my main work.

**Reply by Finii** [reply]
All right. I feared some kind of neovim-GUI application. But then we can concentrate on `Windows Terminal` and font installed directly in Windows (previous comment). Good, this makes it more simple as WSL is not really involved here.

**Comment by Finii** [comment]
See also
* #1165 
* #1063 
* #850

**Comment by Finii** [comment]
Font metric warning...

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/d9188d54-7379-49d3-a57d-9ffff72d37c3)

The cell height looks rather big, but the Powerline glyph is not cut on the rhs.

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/46f9824d-bc0b-41c0-ae5e-be04439362cb)

**Reply by EmreDogann** [reply]
That could just be a weird rendering quirk in Windows Terminal in that case. This is what the glyphs look like in Windows Terminal Preview 1.18.1462.0.
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/484e8ce2-0089-433f-a108-c5712e58e4bf)

Interestingly, the icons are now the correct size, the rounded glyphs aren't cut off anymore but now they seem to be a pixel or so too tall.
With that being said, the issue with the vertical bars is still present in the preview version leading me to believe it's an issue with the font itself.

**Comment by Finii** [comment]
Hmm,

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/f09cb3d0-c5fc-4ab5-96b2-c5ae69d044d4)

We select the bigger line spacing, and that is still not big enough for you? :thinking:

Looking on the originally embedded Powerline glyphs we can determine what they think the call height should be:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/75dfd0a9-e279-4e92-9e38-8b4ab90dd38e)

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/8427500b-d25b-49d2-b6a2-6c7edbd0e62b)

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/1ad2dd1b-46bc-4862-9b3a-d12e8615bb8a)

Looks ok-ish and similar-y, although maybe we are a tad higher. :thinking: But 1200 is for sure too small.

**Reply by EmreDogann** [reply]
> We select the bigger line spacing, and that is still not big enough for you? 🤔

I apologize, but what do you mean by 'select' the bigger line spacing?

This is what report gives me of both the unpatched (top) and patched (bottom) fonts:
```shell
● ~ % font-line report /mnt/c/Fonts/Comic\ Code/ComicCode-Regular.otf | tail -n 15
::::::::::::::::::::::::::::::::::::::::::::::::::
  Baseline to Baseline Distances
::::::::::::::::::::::::::::::::::::::::::::::::::
hhea metrics: 1467
typo metrics: 1200
win metrics:  1467

[OS/2] fsSelection USE_TYPO_METRICS bit set: True

::::::::::::::::::::::::::::::::::::::::::::::::::
  Ratios
::::::::::::::::::::::::::::::::::::::::::::::::::
hhea metrics / UPM:  1.47
typo metrics / UPM:  1.2
win metrics  / UPM:  1.47
● ~ % font-line report /mnt/c/Fonts/Comic\ Code/Nerd\ Font\ Patched/ComicCode-Regular.otf | tail -n 15
::::::::::::::::::::::::::::::::::::::::::::::::::
  Baseline to Baseline Distances
::::::::::::::::::::::::::::::::::::::::::::::::::
hhea metrics: 1467
typo metrics: 1467
win metrics:  1467

[OS/2] fsSelection USE_TYPO_METRICS bit set: True

::::::::::::::::::::::::::::::::::::::::::::::::::
  Ratios
::::::::::::::::::::::::::::::::::::::::::::::::::
hhea metrics / UPM:  1.47
typo metrics / UPM:  1.47
win metrics  / UPM:  1.47
```

**Reply by Finii** [reply]
Strange that you font has a b2b distance of `1467` while the `Comic Code Demo` I just downloaded from MyFonts has `1667` :thinking: 

Whatever.

Sometimes Windows has problems with 'exchanging' fonts. So if you do experiments (or substitude a previously installed original Comic Code with a `--makegroups -1` one), I suggest
* Uninstall old fonts
* Reboot
* Install new fonts

I have seen instances where Windows (Terminal?) seems to cache individual glyphs and you end up with a mix of old and new font.

Regarding your original problem,

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/42ad686a-e7c1-4989-92d0-a54744698368)

The marked thing is the vertical bar you mean, right? It has a small gap to the bar above. Do you know its codepoint (or paste it as text into a comment here)?

**Reply by Finii** [reply]
The bottom at least seems to align correctly

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/f28da629-a5ba-43a2-9deb-683744cdbf7c)

as we neither
* change the cell height
* change the glyph's outline

are you sure the behavior is not already in the unpatched font? Did you check that?

**Reply by EmreDogann** [reply]
> The marked thing is the vertical bar you mean, right? It has a small gap to the bar above. Do you know its codepoint (or paste it as text into a comment here)?

Correct, the code point is U+2502 -> │

**Reply by Finii** [reply]
Ah yes sorry, just reread and noticed this 
![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/954aace2-2314-4e9d-9ede-5da9ced5b9de)

Thank you :-)

**Reply by EmreDogann** [reply]
> are you sure the behavior is not already in the unpatched font? Did you check that?

This is what the unpatched version looks like:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/62ea3cd8-a7a0-4952-aea5-f063f441aada)

**Reply by Finii** [reply]
Ah, the cell is much shorter. Compare with your first image

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/1dee4b91-1a6c-4847-8a77-429618215f47)

The patched font has much more headroom over the lower case L.

The added headroom is where the box drawing glyph (vertical bar) has a gap then, afterwards.

Ok, have a plan, expect a PR soon.

**Reply by EmreDogann** [reply]
ah I see. Looking at CaskaydiaCove the headroom looks correct:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/7a4cd53e-8b03-44a5-9f80-9ad37104d9a6)

May I ask, what is the cause of this extra headroom?

**Reply by Finii** [reply]
> ::::::::::::::::::::::::::::::::::::::::::::::::::
  Baseline to Baseline Distances
::::::::::::::::::::::::::::::::::::::::::::::::::
hhea metrics: 1467
typo metrics: 1200
win metrics:  1467

The font has two different line metrics, HHEA and WIN show the same and TYPO has a smaller cell height.

HHEA is the Apple value, while TYPO and WIN are Microsoft values.
Additionally `USE_TYPO_METRICS bit set: True` so that Windows Terminal uses the TYPO number which is 1200.

The font-patcher makes all 3 metrics identical (because we want to render on all platforms alike), and as you see in the debug output it chooses the WIN value, because it is identical to the HHEA value (which is used on some macOS). So probably that is correct - at least that is the most consistent value. But if the vertical bar is only 1200 high, it will have a gap then :grimacing:

**Reply by Finii** [reply]
`font-line` also reports this for the demo font:

```
[hhea] LineGap:        200
[OS/2] TypoLineGap:    200
```

That means that the demo has an additional height of 200, which is why I have 1667 and you only 1467 (there is no mention of redistributing the gap in your log, so there is no gap). Which is of course strange, but at least the core metrics is identical.

**Comment by Finii** [comment]
Ugg, just noticed `2502` is a box drawing glyph, that should be handled specifically by the patcher.

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/3c5b9155-97e4-4ee8-9bc5-8178f0136363)

There is no mention in your logfile, so they should be untouched.

```python
        box_enabled = self.source_monospaced and not self.symbolsonly # Box glyph only for monospaced and not for Symbols Only
          box_keep = False
          if box_enabled:       
              self.sourceFont.selection.select(("ranges",), 0x2500, 0x259f)
              box_glyphs_target = len(list(self.sourceFont.selection))
              box_glyphs_current = len(list(self.sourceFont.selection.byGlyphs))
              if box_glyphs_target > box_glyphs_current:
                  # Sourcefont does not have all of these glyphs, do not mix sets (overwrite existing)
                  if box_glyphs_current > 0:
                      logger.debug("%d/%d box drawing glyphs will be replaced",
                          box_glyphs_current, box_glyphs_target)
                  box_enabled = True
              else:
                  # Sourcefont does have all of these glyphs
                  # box_keep = True # just scale do not copy (need to scale to fit new cell size)
                  box_enabled = False # Cowardly not scaling existing glyphs, although the code would allow this
```

Ah and the source does have all the box glyphs, so not touching.

**Comment by Finii** [comment]
Are you able to try a `font-patcher` from a branch?

* #1300 

You could then try
* `--metrics TYPO` (keeps the cell size as you see it with the original font)
* `--boxdrawing` (overpatches and scales all boxdrawing glyphs)

(One or the other should be sufficient)

**Reply by EmreDogann** [reply]
Great news, that seems to have solved my issue. Using `--boxdrawing` I get this:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/6e953070-6f20-4496-843b-512fcd943ecf)
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/5d71a5bd-0e52-405d-87f0-29476d0e9627)

The vertical bars are scaled correctly now while keeping the larger line height.
This is great! Thank you so much!

I also tried `--metrics TYPO` which also worked, Windows Terminal incorrectly renders the round glyphs as mentioned before but that's more a Windows Terminal problem I think:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/2b929abb-d095-4680-97bf-a2d7d43143fe)
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/b57c7aa9-e793-431e-9f13-ab9aa6fc2c0d)

With that being said the pre-patched CaskaydiaCove doesn't seem to suffer from that:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/a2756eae-f78d-43f3-b756-151444f87c43)

**Reply by Finii** [reply]
Nice :-)

Which font size do you use, I will try with the demo font then, can be some rounding error :thinking: 

Even more strange, while it is too big top and bottom in your upper image, it is only to big on the top in your middle image.
This smells like rounding or antialiasing.

**Reply by EmreDogann** [reply]
I use font size 12.
All those images were taken from [Windows Terminal Preview 1.18.1462.0](https://github.com/microsoft/terminal/releases/tag/v1.18.1462.0)

**Reply by EmreDogann** [reply]
Hmm I think you're correct on the rounding error/antialiasing. When i zoom in/out via Ctrl+ and Ctrl- it changes from too big only on the top to too big on both bottom and top and so on.

**Reply by Finii** [reply]
What you can try is generating TrueType outlines instead, use `-ext ttf`.
The font internal rounding is different for TrueType or Postscript outlines.
I'm not sure right now what the patcher does, but usually TTF uses integer values in the font file while OTF uses floats.
Well, usually that should not make a visual difference.

Just if you have spare time, you can try that. I'm on the go right now.

**Reply by EmreDogann** [reply]
I tried .ttf but the same issue. I played around a bit more and found out that for CaskaydiaCove, font size 12 and 13 work fine but outside of that it has the same rounding error issues. Similarly for Comic Code, at font size 11 it looks good, but outside of that, it has those rounding issues. This leads me to believe that this is a Windows Terminal issue. What do you think?

CaskaydiaCove, Font Size 12 (13 is also the same):
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/fdf102a5-d339-486d-8dc0-29ccaf807db6)

CaskaydiaCove, Font Size 11:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/e465def3-c69c-4cc0-9846-7d79d35662b8)

Comic Code, Font Size 12:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/27a5028d-212f-4642-9ff6-0b04f142ce15)

Comic Code, Font Size 11:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/52c66170-c21f-4143-940a-ef5f074793b0)

Maybe I should open up an issue over on the Windows Terminal repo about this?

**Reply by Finii** [reply]
Did you try the 'new' rendering engine with Windows Terminal? I have forgotten where one can switch that, I think that is only in the developer version, somewhere.

Btw, it is intended that the Powerline glyphs extend a bit out of the `cell` to avoid dark lines between text-lines, like you see here (but here it is a vertical line):

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/3fbc50d9-21be-4a60-843d-d45d0333f147)

**Reply by EmreDogann** [reply]
Those images above were with AtlasEngine on.

Turning off AtlasEngine (new rendering engine) leads to this:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/8a1f3c09-49e1-4279-970b-7c18b17421a2)

Moreover, when changing font size with AtlasEngine off, it seems the icons are more "stable", meaning the icons are consistently smaller instead of jumping around from the (likely) rounding errors with the AtlasEngine on. I also didn't encounter any dark lines when resizing with AtlasEngine off. However, turning it off also leads to other issues with regard to font icon sizing:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/5151d860-baa9-418f-b33f-762f75663b38)

With AtlasEngine On (Ignore the weird coloring, I haven't set the padding for the icons yet so they're coloured like they're single-width):
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/8a0c2f64-457e-451e-84bc-972270b2376b)

**Reply by EmreDogann** [reply]
So I just tried it in Alacritty, same font and font size:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/99ac4a96-70d1-4121-9689-b50007a60571)

Also interesting the icon colouring seems to be fixed in alacritty as well:
![image](https://github.com/ryanoasis/nerd-fonts/assets/48212096/f82fad61-f1a4-48ad-8df5-548f1cad8200)

I guess with this being a Windows Terminal issue, there's not much that can be done here. Maybe over time it'll get better. Not a huge deal I guess.

Anyway, thanks for your help on this Finii!

**Reply by Finii** [reply]
Thanks for the in depth investigation! That really helps me helping others!

So I guess this is 'fixed' (for some definitions of fixed) when we pull that one PR, right?
Any comments on that PR, is that ok, suggestions etc? :-D

**Reply by EmreDogann** [reply]
> So I guess this is 'fixed' (for some definitions of fixed) when we pull that one PR, right?

Yeah, my main issue was the top headroom on the box glyphs so we can call this "fixed" :D

> Any comments on that PR, is that ok, suggestions etc? :-D

The only suggestion I'd give is regarding documentation. I'd imagine there might be a lot of these cases of problematic fonts. Possibly a new entry in the FAQ/Troubleshooting page talking about this issue and relating the two arguments to address them. Apart from that, everything else is good!
